src/pages/index.js

```diff
@@ -140,26 +167,96 @@ function closeModal(modal) {
 function handleProfileFormSubmit(event) {
   event.preventDefault();
 
-  profileName.textContent = profileModalName.value;
-  profileDescription.textContent = profileModalDescription.value;
+  const submitBtn = event.submitter;
+  setButtonText(submitBtn, true);
+
+  api
+    .editUserInfo({
+      name: profileModalName.value,
+      about: profileModalDescription.value,
+    })
+    .then((data) => {
+      profileName.textContent = data.name;
+      profileDescription.textContent = data.about;
+
+      closeModal(editModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(submitBtn, false);
+    });
+}
+
+function handleAvatarFormSubmit(event) {
+  event.preventDefault();
 
-  closeModal(editModal);
+  const submitBtn = event.submitter;
+  setButtonText(submitBtn, true);
+
+  api
+    .editUserAvatar({ avatar: avatarLink.value })
+    .then((data) => {
+      profileAvatar.src = data.avatar;
+
+      event.target.reset();
+      closeModal(editAvatarModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(submitBtn, false);
+    });
 }
 
 function handleNewPostFormSubmit(event) {
   event.preventDefault();
 
+  const submitBtn = event.submitter;
+  setButtonText(submitBtn, true);
+
   const card = {
     link: newPostLink.value,
     name: newPostName.value,
   };
 
-  renderCard(card);
+  api
+    .addCard(card)
+    .then((cardFromApi) => {
+      renderCard(cardFromApi);
+
+      disableButton(newPostSubmitBtn, true, settings);
+      event.target.reset();
+
+      closeModal(newPostModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(submitBtn, false);
+    });
+}
+
+// Handles Delete Form Submission
+function handleDeleteSubmit(event) {
+  event.preventDefault();
+
+  const submitBtn = event.target;
+
+  if (submitBtn.textContent.trim() === "Delete") {
+    setButtonText(submitBtn, true, "Delete", "Deleting...");
 
-  disableButton(newPostSubmitBtn, true, settings);
-  event.target.reset();
+    api
+      .deleteCard(selectedCardId)
+      .then(() => {
 
-  closeModal(newPostModal);
+        selectedCard.remove();
+        closeModal(deleteModal);
+      })
+      .catch(console.error)
+      .finally(() => {
+        selectedCard = "";
+        selectedCardId = "";
```

Comment:
```js

        selectedCard = "";
        selectedCardId = "";
```

This should be done in a `.then` block to be able to try again in case of a server error


And you should use `undefined` for setting because you've defined the variables as `let selectedCard, selectedCardId;`

```js
        selectedCard = undefined;
        selectedCardId = undefined;
```